File: frontend/src/apis/export.ts

import request from './request'
import { fetchWithTokenRefresh, getAccessToken } from './auth'

/** 导出格式 */
export type ExportFormat = 'markdown'

/** 导出范围 */
export type ExportRange = 'all' | 'recent' | 'custom'

/** 导出请求参数 */
export interface ExportDialogParams {
  format: ExportFormat
  range: ExportRange
  recent_count?: number
  start_index?: number
  end_index?: number
}

/**
 * 导出对话
 * 使用 fetch 直接下载文件（因为 axios 不方便处理 blob 响应）
 */
export const exportDialog = async (dialogId: string, params: ExportDialogParams): Promise<void> => {
  const token = getAccessToken()

  const response = await fetchWithTokenRefresh('/api/v1/export/' + dialogId + '/export', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${token}`
    },
    body: JSON.stringify(params)
  })

  if (!response.ok) {
    const errorData = await response.json().catch(() => ({ detail: '导出失败' }))
    throw new Error(errorData.detail || '导出失败')
  }

  // 从响应头获取文件名
  const disposition = response.headers.get('Content-Disposition') || ''
  let filename = '对话导出'
  const filenameMatch = disposition.match(/filename\*?=UTF-8''(.+?)(?:;|$)/)
  if (filenameMatch) {
    filename = decodeURIComponent(filenameMatch[1])
  } else {
    const simpleMatch = disposition.match(/filename="?(.+?)"?(?:;|$)/)
    if (simpleMatch) {
      filename = simpleMatch[1]
    }
  }

  // 下载文件
  const blob = await response.blob()
  const url = window.URL.createObjectURL(blob)
  const link = document.createElement('a')
  link.href = url
  link.download = filename
  document.body.appendChild(link)
  link.click()
  document.body.removeChild(link)
  window.URL.revokeObjectURL(url)
}

File: backend/app/schemas/export.py

"""对话导出相关Schema"""
from pydantic import BaseModel, Field
from typing import Optional, List
from enum import Enum


class ExportFormat(str, Enum):
    """导出格式"""
    markdown = "markdown"


class ExportRange(str, Enum):
    """导出范围"""
    all = "all"
    recent = "recent"
    custom = "custom"


class ExportDialogRequest(BaseModel):
    """对话导出请求"""
    format: ExportFormat = Field(
        default=ExportFormat.markdown,
        description="导出格式：markdown 或 pdf"
    )
    range: ExportRange = Field(
        default=ExportRange.all,
        description="导出范围：all(全部)、recent(最近N条)、custom(自定义范围)"
    )
    recent_count: Optional[int] = Field(
        default=None,
        description="当range=recent时，指定最近的消息条数"
    )
    start_index: Optional[int] = Field(
        default=None,
        description="当range=custom时，起始消息索引（从0开始）"
    )
    end_index: Optional[int] = Field(
        default=None,
        description="当range=custom时，结束消息索引（不包含）"
    )
